### PR TITLE
9018 fix batch delete

### LINKED ISSUE #9018: Strange behavior when deleting a large number of records
I had a page with 2000 records.
I tried to delete all of them except the 3 most recent ones.
On the UI it seemed like everything went well, I saw 1997 were selected.
Then when I clicked "delete" it started entering a weird mode, I think it tried to fetch the 2000 records locally at first, making my browser slow (but without giving me any feedback that the operation was in progress), then deleting them. 
It seems like the 2000 records were soft-deleted and not 1997 like I asked to.

### PR DESCRIPTION
**Pull request by bosiraphael**
Closes #9018

### CODE REVIEW

File: packages/twenty-front/src/modules/action-menu/components/RecordIndexActionMenu.tsx [hunk 1]
```diff
@@ -25,13 +32,29 @@ export const RecordIndexActionMenu = () => {
 
   const isMobile = useIsMobile();
 
+  const setIsLoadMoreLocked = useSetRecoilComponentStateV2(
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/action-menu/components/RecordIndexActionMenu.tsx:35]: only one componentState to govern all

File: packages/twenty-front/src/modules/action-menu/components/RecordIndexActionMenu.tsx [hunk 2]
```diff
@@ -25,13 +32,29 @@ export const RecordIndexActionMenu = () => {
...
+
   return (
     <>
       {contextStoreCurrentObjectMetadataId && (
         <ActionMenuContext.Provider
           value={{
             isInRightDrawer: false,
-            onActionExecutedCallback: () => {},
+            onActionStartedCallback: (action) => {
+              if (action.key === 'delete-multiple-records') {
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/action-menu/components/RecordIndexActionMenu.tsx:49]: don't we have enum here?

File: packages/twenty-front/src/modules/object-record/hooks/useDeleteManyRecords.ts [hunk 3]
```diff
@@ -112,20 +117,23 @@ export const useDeleteManyRecords = ({
...
+        });
 
-          triggerUpdateRecordOptimisticEffect({
-            cache: apolloClient.cache,
-            objectMetadataItem,
-            currentRecord: cachedRecordWithConnection,
-            updatedRecord: optimisticRecordWithConnection,
-            objectMetadataItems,
-          });
+        triggerUpdateRecordOptimisticEffectByBatch({
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-record/hooks/useDeleteManyRecords.ts:131]: we might want to batch the onError too (30 lines below)

File: packages/twenty-front/src/modules/object-record/record-index/components/RecordIndexContainer.tsx [hunk 4]
```diff
@@ -230,6 +230,7 @@ export const RecordIndexContainer = () => {
               objectNamePlural={objectNamePlural}
               viewBarId={recordIndexId}
             />
+            <RecordIndexTableContainerEffect />
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-record/record-index/components/RecordIndexContainer.tsx:233]: ?

**Review by charlesBochet — APPROVED** [review]
Missing some tests + I think we could remove the non batch optimistic rendering to enforce usage of the new one!

File: packages/twenty-front/src/modules/object-record/record-index/components/RecordIndexContainer.tsx (hunk 4 above)
Comment (bosiraphael) [reply packages/twenty-front/src/modules/object-record/record-index/components/RecordIndexContainer.tsx:233]: Merge error

**Review by charlesBochet — APPROVED** [review]

**Pull request merged**